addons/account/tests/test_account_payment_duplicate.py

```diff
@@ -0,0 +1,74 @@
+from odoo.addons.account.tests.common import AccountTestInvoicingCommon
+from odoo.tests import tagged
+
+
+@tagged('post_install', '-at_install')
+class TestAccountPaymentDuplicate(AccountTestInvoicingCommon):
+    @classmethod
+    def setUpClass(cls, chart_template_ref=None):
+        super().setUpClass(chart_template_ref=chart_template_ref)
+
+        cls.copy_receivable = cls.copy_account(cls.company_data['default_account_receivable'])
+
+        cls.payment = cls.env['account.payment'].create({
+            'amount': 50.0,
+            'payment_type': 'inbound',
+            'partner_type': 'customer',
+            'partner_id': cls.partner_a.id,
+            'destination_account_id': cls.copy_receivable.id,
+        })
+
+    def test_in_payment_single_duplicate_reference(self):
```

Comment:
kinda useless to have a small test doing just that, while the next one do more, if you ask me. 

I'd rather have a single one test method but that makes sure that payments of the same partner, date, amounts but of different type (inbound/outbound) are not detected as duplicates :point_up: